examples/refrigerator-app/silabs/src/RefrigeratorManager.cpp

```diff
@@ -0,0 +1,207 @@
+    // Refrigerator Namespace: 0x41, tag 0x01 (Freezer)
+    constexpr const uint8_t kTagFreezer                                                = 0x01;
+    const Clusters::Descriptor::Structs::SemanticTagStruct::Type refrigeratorTagList[] = { { .namespaceID = kNamespaceRefrigerator,
+                                                                                            .tag         = kTagRefrigerator } };
+    const Clusters::Descriptor::Structs::SemanticTagStruct::Type freezerTagList[]      = { { .namespaceID = kNamespaceRefrigerator,
+                                                                                            .tag         = kTagFreezer } };
+} // namespace
+
+CHIP_ERROR RefrigeratorManager::Init()
+{
+    SetTreeCompositionForEndpoint(kRefEndpointId);
+    SetParentEndpointForEndpoint(kColdCabinetEndpointId, kRefEndpointId);
+    SetParentEndpointForEndpoint(kFreezeCabinetEndpointId, kRefEndpointId);
+
+    // set TagList
+    SetTagList(kColdCabinetEndpointId, Span<const Clusters::Descriptor::Structs::SemanticTagStruct::Type>(refrigeratorTagList));
+    SetTagList(kFreezeCabinetEndpointId, Span<const Clusters::Descriptor::Structs::SemanticTagStruct::Type>(freezerTagList));
+
+    app::Clusters::TemperatureControl::SetInstance(&sAppSupportedTemperatureLevelsDelegate);
+    return CHIP_NO_ERROR;
+}
+
+int8_t RefrigeratorManager::ConvertToPrintableTemp(int16_t temperature)
+{
+    constexpr uint8_t kRoundUpValue = 50;
+
+    // Round up the temperature as we won't print decimals on LCD
+    // Is it a negative temperature
+    if (temperature < 0)
+    {
+        temperature -= kRoundUpValue;
+    }
+    else
+    {
+        temperature += kRoundUpValue;
+    }
+
+    return static_cast<int8_t>(temperature / 100);
+}
+
+
+void RefrigeratorManager::RefAndTempCtrlAttributeChangeHandler(EndpointId endpointId, AttributeId attributeId, uint8_t * value, uint16_t size)
+{
+    switch (attributeId)
+    {
+        case RefAndTempAttr::CurrentMode::Id: {
+            int8_t currentMode = static_cast<int16_t>(*value);
+            mCurrentMode = currentMode;
+        }
+        break;
+
+        case RefAndTempAttr::StartUpMode::Id: {
+            int8_t startUpMode = static_cast<int16_t>(*value);
+            mStartUpMode = startUpMode;
+        }
+        break;
+
+        case RefAndTempAttr::OnMode::Id: {
+            int8_t onMode = static_cast<int16_t>(*value);
+            mOnMode = onMode;
+        }
+        break;
+
+        default: {
+            SILABS_LOG("Unhandled Refrigerator and Temprature attribute %x", attributeId);
+            return;
+        }
+        break;
+    }
+}
+
+void RefrigeratorManager::TempCtrlAttributeChangeHandler(EndpointId endpointId, AttributeId attributeId, uint8_t * value, uint16_t size)
+{
+    switch (attributeId)
+    {
+        case TempCtrlAttr::TemperatureSetpoint::Id: {
+            int16_t temperatureSetpoint = ConvertToPrintableTemp(*((int16_t *) value));
```

Comment:
`ConvertToPrintableTemp(*((int16_t *) value))` is this correct? wont this be `ConvertToPrintableTemp(static_cast<int16_t>(*value))`

Reply:
Done the changes